sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/WhereQuery.cs

```diff
@@ -0,0 +1,238 @@
+
+            whereClauseArg.Append(')');
+            _whereClauses.Add(new WhereClause(whereClauseArg.ToString()));
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a user-specified ADT query language function to check an expression's type against a built type in the ADT query language.
+        /// </summary>
+        /// <param name="expression">The expression that the query is looking for as a specified type.</param>
+        /// <param name="type">The type in the ADT query language being checked for.</param>
+        /// <returns>Query that already contains a SELECT and FROM clause.</returns>
+        public WhereQuery IsOfType(string expression, AdtDataType type)
+        {
+            string functionName = QueryConstants.DataTypeToFunctionNameMap[type];
+            _whereClauses.Add(new WhereClause($"{functionName}({expression})"));
+            return this;
+        }
+
+        /// <summary>
+        /// Used to add nested WHERE conditions to a query.
+        /// </summary>
+        /// <param name="nested">WhereQuery methods to perform within a set of parenthesis.</param>
+        /// <returns>Query that already contains a SELECT and FROM clause.</returns>
+        public WhereQuery Parenthetical(Func<WhereQuery, WhereQuery> nested)
+        {
+            var nestedLogic = new WhereQuery();
+            nested.Invoke(nestedLogic);
+
+            _whereClauses.Add(new WhereClause($"({nestedLogic.GetQueryText()})"));
+            return this;
+        }
+
+        /// <summary>
+        /// Used to add nested WHERE conditions within an AND logical operator to a query.
+        /// </summary>
+        /// <param name="nested">WhereQuery methods to perform within a set of parenthesis.</param>
+        /// <returns>Query that already contains a SELECT and FROM clause.</returns>
+        public WhereQuery And(Func<WhereQuery, WhereQuery> nested)
+        {
+            var nestedLogic = new WhereQuery();
+            nested.Invoke(nestedLogic);
+
+            And();
+            _whereClauses.Add(new WhereClause($"({nestedLogic.GetQueryText()})"));
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the AND logical operator to a query.
+        /// </summary>
+        /// <returns>Query that already contains a SELECT and FROM clause.</returns>
+        public WhereQuery And()
+        {
+            _whereClauses.Add(new WhereClause(QueryConstants.And));
+            return this;
+        }
+
+        /// <summary>
+        /// Used to add nested WHERE conditions within an OR logical operator to a query.
+        /// </summary>
+        /// <param name="nested">WhereQuery methods to perform within a set of parenthesis.</param>
+        /// <returns>Query that already contains a SELECT and FROM clause.</returns>
+        public WhereQuery Or(Func<WhereQuery, WhereQuery> nested)
+        {
+            var nestedLogic = new WhereQuery();
+            nested.Invoke(nestedLogic);
```

Comment:
do we need parenthesis here?

Reply:
I'm assuming you mean the parenthesis around the nested clause when it's injected into the query string? Order of operations-wise, you might be right in that they could be redundant and I can investigate if they can/should be removed